File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Python/Empleado.py

class Empleado:
   
    def __init__(self, nombre: str, edad: int, salario: float):
        self.nombre = nombre
        self.edad = edad
        self.salario = salario

    def __str__(self):
        
        return f"{self.nombre},{self.edad},{self.salario:.2f}"

    def __repr__(self):
        
        return self.__str__()

    @staticmethod
    def from_string(empleado_str: str):
        
        try:
            partes = empleado_str.strip().split(',')
            if len(partes) == 3:
                nombre = partes[0]
                edad = int(partes[1])
                salario = float(partes[2])
                return Empleado(nombre, edad, salario)
            else:
                print(f"Advertencia: Formato de linea incorrecto para empleado: {empleado_str}")
                return None
        except (ValueError, IndexError) as e:
            print(f"Error al parsear linea de empleado '{empleado_str}': {e}")
            return None

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Java/Empleado.java
package Persistencia_de_Objetos.Ejercicio1.Java;

public class Empleado {
    private String nombre;
    private int edad;
    private float salario;

    public Empleado(String nombre, int edad, float salario) {
        this.nombre = nombre;
        this.edad = edad;
        this.salario = salario;
    }

    public String getNombre() {
        return nombre;
    }

    public int getEdad() {
        return edad;
    }

    public float getSalario() {
        return salario;
    }

    @Override
    public String toString() {
        
        return String.format("%s,%d,%.2f", nombre, edad, salario);
    }

    public static Empleado fromString(String empleadoStr) {
        try {
            String[] partes = empleadoStr.trim().split(",");
            if (partes.length == 3) {
                String nombre = partes[0];
                int edad = Integer.parseInt(partes[1]);
                float salario = Float.parseFloat(partes[2]);
                return new Empleado(nombre, edad, salario);
            } else {
                System.err.println("Advertencia: Formato de linea incorrecto para empleado: " + empleadoStr);
                return null;
            }
        } catch (NumberFormatException e) {
            System.err.println("Error al parsear datos numericos de empleado '" + empleadoStr + "': " + e.getMessage());
            return null;
        } catch (IndexOutOfBoundsException e) {
            System.err.println("Error de indice al parsear línea de empleado '" + empleadoStr + "': " + e.getMessage());
            return null;
        }
    }
}
